examples/AvroBlogExamples/Program.cs

```diff
@@ -19,31 +19,32 @@
 using System.IO;
 using System.Linq;
 using System.Threading;
-using Avro;
+using System.Threading.Tasks;
 using Avro.Generic;
 using Confluent.Kafka;
-using Confluent.Kafka.Serialization;
+using Confluent.Kafka.AvroSerdes;
+using Confluent.SchemaRegistry;
 
 
 namespace AvroBlogExample
 {
     /// <summary>
     ///     Complete source for the examples programs presented in the blog post:
-    ///     [insert blog URL here]
+    ///     https://www.confluent.io/blog/decoupling-systems-with-apache-kafka-schema-registry-and-avro/
     /// </summary>
     class Program
     {
-        static void ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
+        async static Task ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
         {
-            var producerConfig = new ProducerConfig { BootstrapServers = bootstrapServers };
+            using (var schemaRegistry = new CachedSchemaRegistryClient(new SchemaRegistryConfig { SchemaRegistryUrl = schemaRegistryUrl }))
+            using (var producer = new Producer(new ProducerConfig { BootstrapServers = bootstrapServers }))
+            {   
+                producer.RegisterAvroSerializer(new AvroSerializer<GenericRecord>(schemaRegistry));

```

Comment:
the way this works kind of abuses the intended purpose of extension methods (refer to the implementation), and there are implications for cleanup (which, with some additional code can be improved upon c.f. the present implementation, and made almost foolproof), but the user experience is great and in practice this will not be an issue (even with the current implementation) since clients are not often created / destroyed.

Reply:
hah! inheritance became completely appropriate and i somehow didn't notice. a tiny refactor replaces the extension methods with `AvroProducer`/`Consumer`s (which are really conceptually `Producer`/`ConsumerWithAvroCapability`), and there are no more cleanup issues.